Update client to ES2022 (#24324)
microsoft/FluidFramework: .changeset/cyan-experts-nail.md

"@fluidframework/routerlicious-urlresolver": minor
"@fluidframework/runtime-definitions": minor
"@fluidframework/runtime-utils": minor
"@fluidframework/sequence": minor
"@fluid-experimental/sequence-deprecated": minor
"@fluidframework/shared-object-base": minor
"@fluidframework/shared-summary-block": minor
"@fluid-experimental/sharejs-json1": minor
"@fluid-private/stochastic-test-utils": minor
"@fluidframework/synthesize": minor
"@fluidframework/task-manager": minor
"@fluidframework/telemetry-utils": minor
"@fluid-private/test-dds-utils": minor
"@fluid-private/test-drivers": minor
"@fluid-private/test-end-to-end-tests": minor
"@fluid-private/test-loader-utils": minor
"@fluid-private/test-pairwise-generator": minor
"@fluidframework/test-runtime-utils": minor
"@fluidframework/test-utils": minor
"@fluid-private/test-version-utils": minor
"@fluidframework/tinylicious-client": minor
"@fluidframework/tinylicious-driver": minor
"@fluidframework/tool-utils": minor
"@fluid-experimental/tree": minor
"@fluidframework/tree": minor
"@fluid-experimental/tree-react-api": minor
"@fluidframework/undo-redo": minor
"__section": other
---
Build packages targeting ES2022

> Packages are now build targeting ES2022.

Reviewer: ```suggestion
Packages are now built targeting ES2022.
```

> This results in reduces bundle size (Measured at 2% for SharedTree) and improves developer experience in debuggers due to JavaScript private fields being visible.

Reviewer: ```suggestion
This results in reduced bundle size (measured at 2% for SharedTree) and improved developer experience in debuggers due to JavaScript private fields being visible.
```

Fluid Framework has not officially supported targets older than ES2022 since before 2.0: this is documented in [ClientRequirements.md](https://github.com/microsoft/FluidFramework/blob/main/ClientRequirements.md) as well as the ReadMe for every client package.
This change does not involve any change to what is officially supported.

> It is possible this change could impact users less up to date JavaScript runtimes:

Reviewer: ```suggestion
It is possible this change could impact users of less up to date JavaScript runtimes:
```
